domain/src/main/java/com/sfaxdroid/domain/GetPixaWallpapersUseCase.kt

```diff
@@ -0,0 +1,33 @@
+package com.sfaxdroid.domain
+
+import com.sfaxdroid.data.entity.PixaTagWithSearchData
+import com.sfaxdroid.data.mappers.PixaItem
+import com.sfaxdroid.data.repositories.ApiService
+import javax.inject.Inject
+import javax.inject.Named
+
+const val pixaBayImageType = "photo"
+const val pixaBayPerPage = "200"
+const val pixaBaySafeSearch = "true"
+
+class GetPixaWallpapersUseCase
+@Inject
+constructor(@Named("pixabay-key") private val pixaBayApiKey: String) {
+    suspend fun getResult(pixaTagWithSearchDataList: List<PixaTagWithSearchData>): List<PixaItem> {
+        var ret = mutableListOf<PixaItem>()
+        pixaTagWithSearchDataList.forEach { s ->
+            val response =
+                ApiService.getInstance()
+                    .getImages(
+                        apiKey = pixaBayApiKey,
+                        category = s.search.category,
+                        imageType = pixaBayImageType,
+                        perPage = pixaBayPerPage,
+                        searchTerm = s.search.searchTerm,
+                        safeSearch = pixaBaySafeSearch
+                    )
+            ret.addAll(response.hits)
+        }
+        return ret
```

Comment:
Use : @Query("category") category: String,

PixaBy Documentation : Filter results by category.
Accepted values: backgrounds, fashion, nature, science, education, feelings, health, people, religion, places, animals, industry, computer, food, sports, transportation, travel, buildings, business, music

pixaTagWithSearchDataList.forEach can block api

Reply:
@query("category") category: String already exists in ApiInterface.kt
![image](https://github.com/yassinBaccour/SfaxDroid_Wallpaper_App/assets/20027626/87b360e8-d79e-4315-ac41-3a31522b5a65)